script/cli/src/libs/explorer.rs

```diff
@@ -8,102 +7,98 @@ use alloy::{
 #[derive(Clone, PartialEq, Eq, Default)]
 pub enum SupportedExplorerType {
     #[default]
+    Manual,
+    EtherscanV2,
     Etherscan,
     Blockscout,
 }
 
 #[derive(Default, Clone)]
-pub struct ExplorerApiLib {
+pub struct Explorer {
     pub name: String,
     pub url: String,
     pub standard: String,
+    pub explorer_type: SupportedExplorerType,
```

Comment:
Recommend having the enum be the top level type with struct variant, to allow for potential of different explorer types needing different parameters / behavior. like

```
enum ExplorerType {
   Etherscan(ExplorerData),
   Blockscout(ExplorerData),
   ...
   NewExplorer(OtherExplorerData),
   
}
```

Reply:
oh that's cool, thanks!